src/api/controllers/auth.controller.js

```diff
@@ -68,6 +82,34 @@ exports.passwordReset = async (req, res, next) => {
 };
 
 /**
+ * Returns success with a valid, existing email + token 
+ * and removes any existing refresh (jwt) and reset tokens
+ * for the user identified by the email.
+ * @public
+ */
+exports.passwordResetChange = async (req, res, next) => {
+  try {
+    // All fields preset and password/confirm match by here
+    const { email, password, confirm, token } = req.body;
+    // Ensure user is valid
+    const user = await User.getByEmail(email);
+    // Find a matching reset token and clear out tokens
+    const valid = await User.matchAndClearAllTokens({ email, token });
+    // Update password for user
+    user.password = password;
+    user.save();
+
+    const userTransformed = user.transform();
```

Comment:
what does `user.transform` do ?

![](https://media.giphy.com/media/GOevs1pZWr4xa/giphy.gif)

Reply:
`transform` is in the User model and part of the original boilerplate I started from. It takes a user object, and basically filters out any system specific fields and cleans it up for returning to the client in reponses.